### PR TITLE
Feature/salesforce earthforce campaign [SF TEMP DEPLOYMENT - DO NOT MERGE]

### PR DESCRIPTION
**Pull request by mohitb35**
- Adds a landing page for the April 2023 Earthforce Fitness Challenge campaign (Salesforce)
- Donate button on landing page links to donate.planet with `utm_campaign` query param in the link.
- Only projects that allow donations are seen in the project grid

Pending:
- [x] leaderboard and tenant score APIs need to be updated

### CODE REVIEW

**Comment by sunilsabatp** [comment]
image

on mobile  version ,currently the marked area is not visible

**Comment by mohitb35** [comment]
> image
> 
> on mobile version ,currently the marked area is not visible

Resolved in https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/1709/commits/3a3c4a11aafd978902822dd5caf9c5dd255046eb

File: src/tenants/salesforce/Campaign/styles/Landing.module.scss [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  align-items: center;
+  background-image: url('/tenants/salesforce/images/bkgd-clouds.png');
+  background-position: top center;
+  background-repeat: no-repeat;
+  background-size: 1440px auto;
+  padding: 35px 0 100px;
+
+  @include smTabletView {
+    margin-top: 0px;
+    // margin-top: 80px;
```
Comment (sunilsabatp) [inline src/tenants/salesforce/Campaign/styles/Landing.module.scss]: remove this comment portion

File: src/tenants/salesforce/Campaign/styles/Landing.module.scss [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  align-items: center;
+  justify-content: center;
+  max-width: 90%;
+  margin-bottom: 10vh;
+  @include smTabletView {
+    max-width: 70%;
+  }
+  @include lgLaptopView {
+    max-width: 1170px;
+    // margin-bottom: 16vh;
```
Comment (sunilsabatp) [inline src/tenants/salesforce/Campaign/styles/Landing.module.scss:41]: remove this comment portion

File: src/tenants/salesforce/Campaign/styles/Landing.module.scss [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    text-align: center;
+    letter-spacing: -0.004em;
+    color: #032d60;
+    margin-bottom: 26px;
+    text-transform: uppercase;
+    @include lgLaptopView {
+      // font-size: $fontXXLarge;
+      line-height: 100%;
+      // margin-bottom: 38px;
+      margin-bottom: 0px;
```
Comment (sunilsabatp) [inline src/tenants/salesforce/Campaign/styles/Landing.module.scss:78]: remove the commented part

**Review by sunilsabatp — CHANGES_REQUESTED** [review]

File: src/tenants/salesforce/Campaign/components/AdditionalInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+          <div className={`${gridStyles.colMd8} ${gridStyles.col12}`}>
+            <p>
+              In January 2020, as a founding partner of 1t.org, Salesforce
+              announced our own goal to support and mobilize the conservation,
+              restoration, and growth of 100 million trees by the end of 2030.
+              In our FY24 Corporate V2MOM, Salesforce aims to reach 50% of this
+              goal - 50 million trees - by year end. This challenge is in
+              support of this initiative all while encouraging greater health
+              and wellness among our amazing employees and providing a chance to
+              earn VTO by exercising for the cause.
```
Comment (sunilsabatp) [inline src/tenants/salesforce/Campaign/components/AdditionalInfo.tsx:25]: is translation needed  here  ?

**Review by sunilsabatp — CHANGES_REQUESTED** [review]

File: pages/vto-fitness-challenge.tsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+  } catch (err) {
+    console.log(err);
+  }
+
+  try {
+    const tenantscoreRes = await fetch(
+      `${process.env.NEXT_PUBLIC_WEBHOOK_URL}/salesforce-earth-month-count`
+    );
+    const tenantScoreArr = await tenantscoreRes.json();
+    console.log(tenantScoreArr);
```
[the remark this answers is not in the document]
Comment (mohitb35) [reply pages/vto-fitness-challenge.tsx:74]: Want the console to be printed to server logs to debug potential issues.

File: src/tenants/salesforce/Campaign/components/ProjectGrid.tsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  return (
+    <div className={`${styles.projectGridContainer}`}>
+      <div className={`${gridStyles.fluidContainer} ${styles.projectGrid}`}>
+        <div
+          className={`${gridStyles.gridRow} ${gridStyles.justifyContentCenter}`}
+        >
+          <div className={`${gridStyles.colMd8} ${gridStyles.col12}`}>
+            <h3>Projects</h3>
+            <p className={styles.contentSectionSubhead}>
+              You can donate to these projects.
```
[the remark this answers is not in the document]
Comment (mohitb35) [reply src/tenants/salesforce/Campaign/components/ProjectGrid.tsx:59]: Translation not needed as the page is in English.

File: pages/vto-fitness-challenge.tsx [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+  return (
+    <>
+      <GetHomeMeta />
+      {initialized ? getCampaignPage() : <></>}
+    </>
+  );
+}
+
+export async function getStaticProps({ locale }) {
```
[the remark this answers is not in the document]
Comment (mohitb35) [reply pages/vto-fitness-challenge.tsx:54]: Done

File: src/tenants/salesforce/Campaign/styles/LeaderBoardSection.module.scss [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+  margin-top: 50px;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  padding-bottom: 20px;
+  position: relative;
+  background-position: center -130px, calc(50% + -370px) 28%,
+    calc(50% + 470px) 48%, calc(50% + 426px) 32%;
+  background-repeat: no-repeat;
+  // background-size: cover, 150px 150px, 67px 67px, 16px 16px;
```
[the remark this answers is not in the document]
Comment (mohitb35) [reply src/tenants/salesforce/Campaign/styles/LeaderBoardSection.module.scss:14]: Done

File: src/tenants/salesforce/Campaign/styles/ProjectGrid.module.scss [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+        margin-bottom: 15px;
+      }
+    }
+    p {
+      font-family: $primaryFontFamily;
+      font-style: normal;
+      font-weight: normal;
+      font-size: 15px;
+      line-height: 24px;
+      // color: #ffffff;
```
[the remark this answers is not in the document]
Comment (mohitb35) [reply src/tenants/salesforce/Campaign/styles/ProjectGrid.module.scss:68]: Done

File: src/tenants/salesforce/Campaign/styles/ProjectGrid.module.scss [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+
+  p.contentSectionSubhead {
+    font-family: $primaryFontFamily;
+    font-style: normal;
+    font-weight: 300;
+    font-size: $fontMedium;
+    line-height: 25px;
+    text-align: center;
+    letter-spacing: -0.002em;
+    // color: #ffffff;
```
[the remark this answers is not in the document]
Comment (mohitb35) [reply src/tenants/salesforce/Campaign/styles/ProjectGrid.module.scss:109]: Done

File: src/tenants/salesforce/Campaign/index.tsx [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+  tenantScore: { total: number };
+}
+
+export default function Campaign({ tenantScore, leaderboard }: Props) {
+  return (
+    <>
+      <Head>
+        <title>{`VTO Fitness Challenge | ${config.meta.title}`}</title>
+      </Head>
+      <main style={{ backgroundColor: 'white', paddingBottom: '60px' }}>
```
[the remark this answers is not in the document]
Comment (mohitb35) [reply src/tenants/salesforce/Campaign/index.tsx:27]: Left as is. The changes in this PR are temporary and will be removed once the campaign is over. They will also not be merged to main.

File: src/tenants/salesforce/Campaign/styles/ContentSection.module.scss [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+    padding-bottom: 150px;
+    h2 {
+      font-family: $secondaryFontFamily;
+      font-style: normal;
+      font-weight: 500;
+      font-size: $fontXXLarge;
+      line-height: 40px;
+      text-align: center;
+      letter-spacing: -0.004em;
+      // color: #ffffff;
```
[the remark this answers is not in the document]
Comment (mohitb35) [reply src/tenants/salesforce/Campaign/styles/ContentSection.module.scss:22]: Done

File: src/tenants/salesforce/Campaign/styles/ContentSection.module.scss [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+
+  p.contentSectionSubhead {
+    font-family: $primaryFontFamily;
+    font-style: normal;
+    font-weight: 300;
+    font-size: $fontMedium;
+    line-height: 25px;
+    text-align: center;
+    letter-spacing: -0.002em;
+    // color: #ffffff;
```
[the remark this answers is not in the document]
Comment (mohitb35) [reply src/tenants/salesforce/Campaign/styles/ContentSection.module.scss:84]: Done

File: src/tenants/salesforce/Campaign/styles/Grid.module.scss [hunk 14]
```diff
@@ -0,0 +1,189 @@
...
+}
+
+.illustration2 {
+  width: 256px;
+  max-width: 100%;
+  height: auto;
+}
+
+.mb65100 {
+  // margin-bottom: 65px;
```
[the remark this answers is not in the document]
Comment (mohitb35) [reply src/tenants/salesforce/Campaign/styles/Grid.module.scss:38]: Done

File: src/tenants/salesforce/Campaign/styles/Landing.module.scss (hunk 1 above)
Comment (mohitb35) [reply src/tenants/salesforce/Campaign/styles/Landing.module.scss]: Done

File: src/tenants/salesforce/Campaign/styles/Landing.module.scss (hunk 2 above)
Comment (mohitb35) [reply src/tenants/salesforce/Campaign/styles/Landing.module.scss:41]: Done

File: src/tenants/salesforce/Campaign/styles/Landing.module.scss (hunk 3 above)
Comment (mohitb35) [reply src/tenants/salesforce/Campaign/styles/Landing.module.scss:78]: Done

**Review by sunilsabatp — APPROVED** [review]

**Pull request closed without merging**